packages/@aws-cdk/aws-route53/lib/heath-check/alarm-health-check.ts

```diff
@@ -0,0 +1,80 @@
+import cloudwatch = require('@aws-cdk/aws-cloudwatch');
+import { Aws, Construct } from '@aws-cdk/core';
+import { AdvancedHealthCheckOptions, HealthCheck } from "./health-check";
+
+/**
+ * Alarm health check properties
+ * @experimental
+ */
+export interface AlarmHealthCheckProps extends AdvancedHealthCheckOptions {
+    /**
+     * The CloudWatch alarm to be monitored
+     *
+     * Supported alarms:
+     * * Standard-resolution metrics (High-resolution metrics aren't supported)
+     * * Statistics: Average, Minimum, Maximum, Sum, and SampleCount
+     *
+     * Route 53 does not support alarms that use metric math to query multiple CloudWatch metrics.
+     */
+    readonly alarm: cloudwatch.Alarm;
+    /**
+     * Status of the health check when CloudWatch has insufficient data to determine
+     * the state of the alarm that you chose for CloudWatch alarm
+     *
+     * @default InsufficientDataHealthStatus.LAST_KNOWN_STATUS
+     */
+    readonly insufficientDataHealthStatus?: InsufficientDataHealthStatusType;
+}
+
+/**
+ * Alarm health check construct
+ *
+ * @resource AWS::Route53::HealthCheck
+ * @see https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/health-checks-creating-values.html#health-checks-creating-values-cloudwatch
+ * @experimental
+ */
+export class AlarmHealthCheck extends HealthCheck {
+    public constructor(scope: Construct, id: string, props: AlarmHealthCheckProps) {
+        const { alarm, ...baseProps } = props;
+        super(scope, id, {
+            type: AlarmHealthCheckType.CALCULATED,
+            ...baseProps,
+            alarmIdentifier: {
+                name: props.alarm.alarmName,
+                // FIXME not always true?
+                region: Aws.REGION,
```

Comment:
I don't know how to retrieve the alarm's region there

Reply:
I tried writing the following test:

```ts
// GIVEN
const app = new App();
const stack1 = new Stack(app, 'Stack1', {env: {region: 'us-east-1'}});
const stack2 = new Stack(app, 'Stack2', {env: {region: 'eu-west-1'}});

const alarm = new cloudwatch.Alarm(stack1, 'Alarm', {
  metric: new cloudwatch.Metric({
    metricName: 'Errors',
    namespace: 'my.namespace',
  }),
  threshold: 1,
  evaluationPeriods: 1,
});

// WHEN
new route53.AlarmHealthCheck(stack2, 'HealthCheck', {alarm});
```

But I'm getting the following error:
`Resolution error: Resolution error: Resolution error: Cannot use resource 'Stack1/Alarm' in a cross-environment fashion, the resource's physical name must be explicit set or use `PhysicalName.GENERATE_IF_NEEDED`.`

It this expected?